regression/goto-analyzer/value-set-function-pointers-simple/test.desc

```diff
@@ -10,7 +10,7 @@ main.c
 ^main::1::fun1 \(\) -> value-set-begin: ptr ->\(f\) :value-set-end
 ^main::1::fun2_show \(\) -> value-set-begin: ptr ->\(f\), ptr ->\(g\) :value-set-end
 ^main::1::fun3_show \(\) -> value-set-begin: ptr ->\(f\), ptr ->\(g\) :value-set-end
-^fun_global_show \(\) -> value-set-begin: ptr ->\(f\), ptr ->\(g\) :value-set-end
+^fun_global_show \(\) -> value-set-begin: TOP :value-set-end
```

Comment:
Looking at the test case this is slightly concerning as `{f, g}` is a legit and reasonable thing for it to be and it should only be set to `TOP` if something really wrong is occurring.  This looks like we might need to be a bit careful about how this PR interacts with the abstract interpreter.